### ISSUE TITLE
[iOS][Barcode Scanner] zXing handled barcode types are draining battery

### ISSUE THREAD

**Issue by tgjorgoski** [expo/expo#6804]
How to reproduce:
Just add a Barcode scanner, and keep it open in iOS.
The phone warms up, and you can watch battery draining in real time...

I was looking through the pull requests, and possibly it is a bug introduced by:
https://github.com/expo/expo/pull/5976 ?

If so, can we please have an option to use the native iOS scanning?
The way it is now, it is preventing us from actually shipping our app. At the rate the battery is drained, it really makes the app unusable.

**Comment by tgjorgoski** [comment]
OK, once I looked into the PR, we solved the issue by just limiting the types of barcodes that we scan to the ones that we use. 
Now the problem with the scanning is gone. 
You might still want to update the documentation though, to clear up that if certain types are used, the battery on iOS will drain.

**Comment by cruzach** [comment]
@tgjorgoski thanks for pointing this out, and investigating more on your own!

I can update the docs, did you notice any specific barcode types that ended up draining battery faster than others?

**Comment by tgjorgoski** [comment]
Didn't do such a deep investigation, sorry, we just limited the types to QR code. ( Note also, that the draining rate is definitely reduced , but I can't say for sure that maybe it is still spending more CPU than previously). 

Maybe @bbarthec can put more light on it?

**Comment by bbarthec** [comment]
@tgjorgoski, this battery draining scenario is definitely a problematic thing and thank you very much for pointing this out 🙇 probably there are some optimisations that could be done to limit this problem 🤔
@cruzach, at the moment I suspect all the barcode types that are excluded from native iOS scanning process and instead handled by newly added `zxing` flow (#5976) would trigger this battery draining issue. The types that definitely would be problematic are listed (in these lines)[https://github.com/expo/expo/blob/1f9c336a8f474a80b12030c7c450cc7f510a01b2/packages/expo-barcode-scanner/ios/EXBarCodeScanner/EXBarCodeScanner.m#L38-L43].

**Comment by Gogn** [comment]
I'm also faced with the issue.

After a screen with expo-barcode-scanner was open the app use a lot of CPU, even after the screen with expo-barcode-scanner was closed.

"expo": "~44.0.
"expo-barcode-scanner": "~11.2.0",

Screenshot 1: the screen with expo-barcode-scanner wasn't open:
![Screenshot 2022-03-17 at 19 19 05](https://user-images.githubusercontent.com/1881637/158840549-c10c3fa7-d203-46f7-9447-279a26b98186.jpg)


Screenshot 2: the screen with expo-barcode-scanner was open and then another screen was opened:
![Screenshot 2022-03-17 at 19 20 09](https://user-images.githubusercontent.com/1881637/158840575-0b7174fc-732a-44ef-b16c-c1eb866bbbb4.jpg)

Screenshot 3: the screen with expo-barcode-scanner with param "barCodeTypes={[BarCodeScanner.Constants.BarCodeType.qr]}" was open and then another screen was opened:
![Screenshot 2022-03-17 at 19 19 36](https://user-images.githubusercontent.com/1881637/158840580-6fb7b898-06b1-4b4c-8c51-a856d410bd55.jpg)

Param "barCodeTypes={[BarCodeScanner.Constants.BarCodeType.qr]}" allow to reduce CPU usage but it doesn't stop when switching to another window (via react-navigation)

**Comment by JakeCroz1** [comment]
I have the same issue with the Expo Barcode Scanner API draining the battery fast. I specified the barcode type but the battery still drains pretty fast. Also, did notice that this API still runs in the background for a few seconds after the app is closed. I do not see how this Barcode Scanner API is a viable solution in a production app with this existing issue. I submitted a bug issue highlighting this problem but it seems like it taking forever to get reviewed. Are expo developers just ignoring this issue?
